### PR TITLE
Refactor mute and vote mute code

### PR DESCRIPTION
**Pull request by ChillerDragon**
Fix enum name for VOTE_MUTES break out earlier in mutes loop. And using goto to make the code more readable o.O

### CODE REVIEW

**Comment by def-** [comment]
How about another function instead of goto?

**Comment by ChillerDragon** [comment]
I don't see why another function could make this nicer. I can only imagine a function that takes similar amount of arguments and even returns something. It would make readability worse not better imo.

**Comment by def-** [comment]
```
bool Success = MuteStuffImpl(...);
if(Success)
{
    // write your code here instead of "goto success;"
}
```

And in MuteStuffImpl instead of `goto success` use `return true`.

**Comment by ChillerDragon** [comment]
> ```
> bool Success = MuteStuffImpl(...);
> if(Success)
> {
>     // write your code here instead of "goto success;"
> }
> ```
> And in MuteStuffImpl instead of `goto success` use `return true`.

ikr but thats what I mean. To understand the full logic you have to scroll somewhere to see the contect of the function. You have to pass in arguments that are similar to the outer function.

Any reason why not to use goto?

**Comment by def-** [comment]
goto is harder to reason about. If you really really didn't want to call another function, lambdas. But that doesn't seem Teeworld-ish.

File: src/game/server/ddracecommands.cpp [hunk 1]
```diff
@@ -345,44 +345,40 @@ void CGameContext::ConForcePause(IConsole::IResult *pResult, void *pUserData)
...
+	if(pDisplayName)
 	{
-		if(pDisplayName)
-		{
-			str_format(aBuf, sizeof aBuf, "'%s' banned '%s' for %d seconds from voting.",
-				Server()->ClientName(AuthedID), pDisplayName, Secs);
-			Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "votemute", aBuf);
-		}
+		char aBuf[128];
+		str_format(aBuf, sizeof aBuf, "'%s' banned '%s' for %d seconds from voting.",
```
Comment (12pm) [inline src/game/server/ddracecommands.cpp]: Don't announce who did the vote ban

**Comment by Learath2** [comment]
Goto's are not inherently evil, but in this use case they create spaghetti code. Jumping over that return makes it harder to ascertain where the flow is going.

But in this case I'd argue that you shouldn't even bother announcing updates to mutes, just return from there IMHO.

**Comment by ChillerDragon** [comment]
@Learath2 @12pm Imo this action should be visible to the server. Staff should be responsible for what they do. If a mod vote mutes a troll the server knows who is the hero and fixed it. And if a mod does random vote muting players know where to ask for reasons.

**Comment by 12pm** [comment]
It's inconsistent with the other commands and I'm not sure if we want this

**Comment by def-** [comment]
bors r+
thanks

**Comment by def-** [comment]
bors r-
Wait, I didn't see jao's comment.

**Comment by Ryozuki** [comment]
Imho mods actions should be public.

**Comment by def-** [comment]
Then add it everywhere and not just in one action. But do so after a discussion and in a separate change.

**Comment by ChillerDragon** [comment]
@def- i didn't change it tho vote mute has always been like this.

**Pull request merged**